src/types/llm.ts

```diff
@@ -0,0 +1,53 @@
+import { GROUND_TRUTHS_SYSTEM_MESSAGES } from "../handlers/ground-truths/prompts";
+
+export type ModelApplications = "code-review" | "chat-bot";
+
+type ChatBotAppParams = {
+  languages: [string, number][];
+  dependencies: Record<string, string>;
+  devDependencies: Record<string, string>;
+};
+
+type CodeReviewAppParams = {
+  taskSpecification: string;
+};
+
+export type AppParamsHelper<TApp extends ModelApplications> = TApp extends "code-review"
+  ? CodeReviewAppParams
+  : TApp extends "chat-bot"
+    ? ChatBotAppParams
+    : never;
+
+export type CompletionsModelHelper<TApp extends ModelApplications> = TApp extends "code-review" ? "gpt-4o" : TApp extends "chat-bot" ? "o1-mini" : never;
```

Comment:
I implemented this thinking it'd be too slow but I think `o1-mini` is fast enough actually. If this is not helpful I'll remove it. 

1. This enforces the correct usage at the type level.
2. Hardcoded but might be able to make it dynamic if required.

Reply:
The suggestion is to use claude when parsing code as was recommended to me by @0x4007 however `o1-mini` will probably be used instead as it's proven quick enough so far in my testing.

In #11 we are passing the entire PR diff + the task spec which has had truths generated from it also that exist within the system message. Are you guys suggesting that I pass the spec + diff to claude instead of GPT which by my understanding is the more advanced model?

I ask because it's not just parsing the code, there is a very complex degree of reasoning etc required to be able to tie the changes to the spec effectively. Is Claude sufficient for such large tasks?

Reply:
What should be fed to Claude is my base question: just code? or code and any other context and requests that we'd ask of GPT?

Reply:
So to clarify, for #11, the context window will contain:

- raw task spec
- ground truths sourced from the task spec
- code review prompt/instructions
- raw PR diff

These should all be fed to Claude 3.5 instead of GPT, and Claude will be performing our code reviews (comparing spec to changes made, not writing code) start-to-finish (minus ground-truth sourcing)?


> I think it would be best if any information related to a coding task is passed to Claude, while we default to the other OpenAI model for everything else.

Your usage of task here confuses me sorry. All of our github issues are coding tasks. Do you mean any queries that involve raw code should be fed to Claude instead of GPT?

Reply:
Okay this will need another task opened to alter the `@ubqbot` logic as all context goes to `o1-mini` including pull request diffs currently.

Would you like to make a task for it?

Reply:
In my deep experience, when I am building off of and integrating/editing existing code I always use Claude. It's careful and considerate of existing code. But if I'm one-shotting a brand new feature that doesn't import/edit existing code I'll use o1-preview. It's really good at being its own leader where Claude is much better as an "assistant" 

If I need to pass in some code context but it's not directly being used/imported in the newly generated code, I use o1-mini because it has a larger usable context window (both preview and mini have ~65k tokens per request but preview eats up most of them for internal reasoning steps)

In conclusion, when dealing with specific details of code, vs the "general idea of it" always use Claude. It's far more precise. 

Hope that helps.

Reply:
`@ubqbot` because it's pulling in context from multiple task specs, multiple convos and PRs (raw code) I think GPT is also more appropriate but I've never used Claude.

---

By my understanding of the feature introduced in #11 we need it to be able to compartmentalize the spec and the code and make very accurate deductions regarding it's implementation, like your description of `o1` being it's own leader.

nuances with code review:
- specs may be lacking and/or hard to extrapolate from, requires in-prod testing
- we need it to enforce the spec but not to the letter as it'll bounce PRs for too many false flags (as I've witnessed building the last review logic)

Reply:
Resolving this thread as this requires it's own task as there is a lot of discussion and potential work involved in categorizing queries on the fly and determining if they should be fed to Claude or GPT. It should not hold back this PR.